src/modules/map/components/cesium/utils/addImageryLayer-mixins.js

```diff
@@ -0,0 +1,63 @@
+const addImageryLayerMixins = {
```

Comment:
a little (big) docs here is wanted, to explain what is expected from a component using this mixin

You seem to be wanting a method instead of a property, is that something you ended up because just requiring `this.layer` was not working as intended? Otherwise I would prefer we stick to creating a `this.layer = new ...` in `beforeCreate()` hooks in the components using the mixin, and the mixin is responsible for mounting and unmounting it from Cesium (the component creating it is responsible for its deletion)

Reply:
I added doc

I used the method because there are two places where the layer creates, when adding and when the URL changes (e.g. year changed). Also, WMTS and WMS are using different providers so we can't have it in mixin